hybrid/battery.py

```diff
@@ -255,6 +259,9 @@ def simulate_financials(self, interconnect_kw: float, project_life: int, cap_cre
         :param cap_cred_avail_storage: Base capacity credit on available storage (True),
                                             otherwise use only dispatched generation (False)
         """
+        if not isinstance(self._financial_model, Singleowner.Singleowner):
+            self._financial_model.assign(self._system_model.export(), ignore_missing_vals=True)       # copy system parameter values having same name
+        
```

Comment:
This derived class implementation of `simulate_financials` is overwriting the base class implementation in `power_source`, with no call to the base class function. So I think it is less complicated as-is.

Reply:
Hmm.. I'm starting to get second thoughts regarding the necessity of this function, to copy over system parameter values with the same name. First off, it does a lot of unnecessary work since most if not all of the system parameter values will not be needed, so it's a bit too much of a shotgun approach. I prefer having just a requirement for the `set_financial_inputs` function since that's super clear what needs to happen.

Reply:
What do you think?

Reply:
I agree that it's a shotgun approach with potential for unwanted linkage between system and custom financial parameters. I wouldn't think the unnecessary work is significant in the simulation time, but I don't know. I'd say overall you're probably right with removing the assign statement and expanding `set_financial_inputs` to include all the variables that should be linked to the system model.

Reply:
Thanks!